surrealdb/surrealdb_test.go

```diff
@@ -0,0 +1,253 @@
+package surrealdb
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/surrealdb"
+	"os"
+	"testing"
+	"time"
+)
+
+var (
+	// surrealDb is the default image used for running surrealdb in tests.
+	surrealDb                   = "surrealdb/surrealdb:latest"
+	surrealDbImageEnvVar string = "TEST_SURREALDB_IMAGE"
+	surrealDbUser        string = "root"
+	surrealDbPass        string = "root"
+)
+
+func newTestStore(t testing.TB) (*Storage, error) {
```

Comment:
@aliyasirnac may I take over the testcontainers code and add changes on top of yours? I don't want you to feel I'm too picky and convert this in infinite ping-pong 🏓  causing your frustration 🙏

Reply:
All storage modules are defining this `newTestStore` function just returning a pointer to `Storage`, using require.Error as part of the internals of the helper function. Could you apply that, updating all references? Each test function will call it, and will defer the closing of the returned store in the test function, so it's clear for eventual readers that each test function closes the storage. There is one single exception, the Test function for closing a storage, which does not defer it but simply checks it can be closed without errors. Please take a look at the recently added redis or arangodb tests 🙏 

Again, thanks for your work here

Reply:
@aliyasirnac I think this is the only thing missing, everything else looks good to me 💪